server/api/src/rsDataObjGet.cpp

```diff
@@ -82,6 +82,16 @@ namespace
                 __FUNCTION__, __LINE__, bytes_read,
                 replica.logical_path(), replica.hierarchy()));
         }
+        else if (0 == bytes_read) {
+            // Because clients may expect a buffer to always be returned, we set the
+            // length of the returned buffer to zero to indicate that it doesn't contain
+            // any valid data. The real solution is to not allocate anything and the clients
+            // use the bytes read as the indicator for how to move forward.
```

Comment:
Are we going to file an issue to implement the "real solution"?

Reply:
I should probably reword that.

Allocating a buffer isn't that big of a deal. It's not wrong, just inefficient. If we made this change, we may break clients (which have a bug if they are affected by this PR). Is the real solution a 4.3 or 5.0 thing?

Reply:
Sounds like maybe a 5.0 thing.

Reply:
I think we're okay for this PR to be 42s - oh I see... this is about the wording of a 'real' solution in the comment...

Reply:
i think @SwooshyCueb is right.

this PR is good though